collector/netframework_clrmemory.go

```diff
@@ -0,0 +1,388 @@
+// returns data points from Win32_PerfRawData_NETFramework_NETCLRMemory
+// <add link to documentation here> - Win32_PerfRawData_NETFramework_NETCLRMemory class
+package collector
+
+import (
+	"log"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	Factories["netframework_clrmemory"] = NewNETFramework_NETCLRMemoryCollector
+}
+
+// A NETFramework_NETCLRMemoryCollector is a Prometheus collector for WMI Win32_PerfRawData_NETFramework_NETCLRMemory metrics
+type NETFramework_NETCLRMemoryCollector struct {
+	AllocatedBytesPersec               *prometheus.Desc
+	FinalizationSurvivors              *prometheus.Desc
+	Gen0heapsize                       *prometheus.Desc
+	Gen0PromotedBytesPerSec            *prometheus.Desc
+	Gen1heapsize                       *prometheus.Desc
+	Gen1PromotedBytesPerSec            *prometheus.Desc
+	Gen2heapsize                       *prometheus.Desc
+	LargeObjectHeapsize                *prometheus.Desc
+	NumberBytesinallHeaps              *prometheus.Desc
+	NumberGCHandles                    *prometheus.Desc
+	NumberGen0Collections              *prometheus.Desc
+	NumberGen1Collections              *prometheus.Desc
+	NumberGen2Collections              *prometheus.Desc
+	NumberInducedGC                    *prometheus.Desc
+	NumberofPinnedObjects              *prometheus.Desc
+	NumberofSinkBlocksinuse            *prometheus.Desc
+	NumberTotalcommittedBytes          *prometheus.Desc
+	NumberTotalreservedBytes           *prometheus.Desc
+	PercentTimeinGC                    *prometheus.Desc
+	PromotedFinalizationMemoryfromGen0 *prometheus.Desc
+	PromotedMemoryfromGen0             *prometheus.Desc
+	PromotedMemoryfromGen1             *prometheus.Desc
+}
+
+// NewNETFramework_NETCLRMemoryCollector ...
+func NewNETFramework_NETCLRMemoryCollector() (Collector, error) {
+	const subsystem = "netframework_clrmemory"
+	return &NETFramework_NETCLRMemoryCollector{
+		AllocatedBytesPersec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "allocated_bytes_persec"),
+			"Displays the number of bytes per second allocated on the garbage collection heap.",
+			[]string{"process"},
+			nil,
+		),
+		FinalizationSurvivors: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "finalization_survivors"),
+			"Displays the number of garbage-collected objects that survive a collection because they are waiting to be finalized.",
+			[]string{"process"},
+			nil,
+		),
+		Gen0heapsize: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "gen0heap_size_bytes"),
+			"Displays the maximum bytes that can be allocated in generation 0; it does not indicate the current number of bytes allocated in generation 0.",
+			[]string{"process"},
+			nil,
+		),
+		Gen0PromotedBytesPerSec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "gen0_promoted_bytes_per_sec"),
```

Comment:
This and the corresponding Gen1 are per_sec-style counters that also need new names, probably `gen0_promoted_bytes_total`?

Reply:
Yes simply removed the "per_sec" suffix to keep it coherent with what I choosed with the similar perf counters